python/cudf/cudf/core/multiindex.py

```diff
@@ -1636,72 +1637,50 @@ def _level_index_from_level(self, level):
-        ...     [('a', 'd'), ('b', 'e'), ('b', 'f')])
-        >>> mi.get_loc('b')
-        slice(1, 3, None)
-        >>> mi.get_loc(('b', 'e'))
-        1
-        >>> non_monotonic_non_unique_idx = cudf.MultiIndex.from_tuples(
-        ...     [('c', 'd'), ('b', 'e'), ('a', 'f'), ('b', 'e')])
-        >>> non_monotonic_non_unique_idx.get_loc('b') # differ from pandas
-        slice(1, 4, 2)
-
-        .. pandas-compat::
-            **MultiIndex.get_loc**
-
-            The return types of this function may deviates from the
-            method provided by Pandas. If the index is neither
-            lexicographically sorted nor unique, a best effort attempt is made
-            to coerce the found indices into a slice. For example:
-
-            .. code-block::
-
-                >>> import pandas as pd
-                >>> import cudf
-                >>> x = pd.MultiIndex.from_tuples([
-                ...     (2, 1, 1), (1, 2, 3), (1, 2, 1),
-                ...     (1, 1, 1), (1, 1, 1), (2, 2, 1),
-                ... ])
-                >>> x.get_loc(1)
-                array([False,  True,  True,  True,  True, False])
-                >>> cudf.from_pandas(x).get_loc(1)
-                slice(1, 5, 1)
-        """
+    def get_indexer(self, target, method=None, limit=None, tolerance=None):
         if tolerance is not None:
             raise NotImplementedError(
                 "Parameter tolerance is not supported yet."
             )
-        if method is not None:
+        if method == "nearest":
             raise NotImplementedError(
-                "only the default get_loc method is currently supported for"
-                " MultiIndex"
+                f"{method=} is not supported yet for MultiIndex."
             )
 
+        target = cudf.MultiIndex.from_tuples(target)
+        needle_table = target.to_frame(index=False)
+        col_names = list(range(0, self.nlevels))
+        needle_table["order"] = needle_table.index
+        haystack_table = self.copy(deep=True).to_frame(index=False)
+        haystack_table["order"] = haystack_table.index
+        merged_table = haystack_table.merge(
+            needle_table, on=col_names, how="outer"
+        )
+        result_series = (
+            merged_table.sort_values(by="order_y")
+            .head(len(target))["order_x"]
+            .reset_index(drop=True)
+        )
+        if method is None:
+            result_series = result_series.fillna(-1)
+        elif method in {"ffill", "bfill", "pad", "backfill"}:
+            result_series = _get_indexer_basic(
+                index=self,
+                positions=result_series,
+                method=method,
+                target_col=needle_table[col_names],
+                tolerance=tolerance,
+            )
+        else:
+            raise ValueError(
+                f"{method=} is unsupported, only supported values are: "
+                f"{['ffill', 'bfill', None]}"
```

Comment:
Same as above -- pad, backfill appear to be supported.

Reply:
Fixed 👍